### ISSUE TITLE
Windows spawned in non-visible workspaces should not get focus

### ISSUE THREAD

**Issue by carrascomj** [leftwm/leftwm#760]
**Describe the bug**
When a window rule makes a window appear in a non-visible workspace - that is, in a workspace which is not being displayed in any monitor - the user loses focus of the current window.

**To Reproduce**
1. Add this to your config.

```toml
window_rules = [ 
  {window_class = "Navigator", spawn_on_tag = 8},
]

[[keybind]]
command = "Execute"
value = "firefox"
modifier = ["modkey"]
key = "b"
```
3. Put all monitors in workspaces that are not the 8th one.
4. Reload `leftwm`.
5. Open a window (wait for focus) and press Super-B.
6. You have lost focus of the window.

**Expected behavior**
The user should not lose focus. Possible configurations would include to be able to jump the workspace where the window opened instead but I don't think the focus should be lost to a non-visible window in any case.

**Environment:**
 - OS/Distro: 5.17.4-arch1-1
 - Output of `leftwm-check`

```
:: LeftWM version: 0.3.0
:: LeftWM git hash: c0fcf84
:: Loading configuration . . .
    -> Configuration loaded OK 
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK 
:: Checking theme . . .
    -> Theme O
```

**Comment by carrascomj** [comment]
I am using _workspace_ when I wanted to say _tag_ in some cases. I hope you get the gist :sweat:.

**Comment by Zwo1in** [comment]
So I just discovered DisplayActions :upside_down_face:. It looks like during error case, leftwm receives an extra SendToTag event after it already has sent a window to tag 3.

Log containing also leftwm acttions (a bit shortened):
```
Got next events from display server: [
    WindowCreate(
        Window {
            FIREFOX WINDOW
        },
        960,
        540,
    ),
]
Executing action: AddedWindow(
    XlibHandle(
        FIREFOX WINDOW,
    ),
    false,
    false,
)
Executing action: SetWindowTags(
    XlibHandle(
        FIREFOX WINDOW,
    ),
    [
        3,
    ],
)
Executing action: SetWindowOrder(
    [
        Window {
            ALACRITTY WINDOW
        },
        Window {
            FIREFOX WINDOW
        },
        Window {
            CONKY
        },
        Window {
            POLYBAR
        },
    ],
)
Got next events from display server: [
    WindowChange(
        WindowChange {
            handle: XlibHandle(
                FIREFOX WINDOW,
            ),
            transient: None,
            never_focus: None,
            name: None,
            type: None,
            floating: None,
            strut: None,
            requested: None,
            states: Some(
                [
                    MaximizedVert,
                    MaximizedHorz,
                ],
            ),
        },
    ),
]
Got next events from display server: []
Got next events from display server: []
Got next events from display server: [
    SendCommand(
        SendWindowToTag {
            window: Some(
                XlibHandle(
                    FIREFOX WINDOW,
                ),
            ),
            tag: 3,
        },
    ),
]
Executing action: SetWindowTags(
    XlibHandle(
        FIREFOX WINDOW,
    ),
    [
        3,
    ],
)
Executing action: SetWindowOrder(
    [
        Window {
            ALACRITTY WINDOW
        },
        Window {
            FIREFOX WINDOW
        },
        Window {
            CONKY
        },
        Window {
            POLYBAR
        },
    ],
)
Got next events from display server: []
Got next events from display server: []
Got next events from display server: []
Got next events from display server: []
Got next events from display server: []
Got next events from display server: []
Got next events from display server: [
    WindowChange(
        WindowChange {
            handle: XlibHandle(
                FIREFOX WINDOW,
            ),
            transient: None,
            never_focus: Some(
                false,
            ),
            name: None,
            type: None,
            floating: None,
            strut: None,
            requested: None,
            states: None,
        },
    ),
    ConfigureXlibWindow(
        XlibHandle(
            FIREFOX WINDOW,
        ),
    ),
    HandleWindowFocus(
        XlibHandle(
            FIREFOX WINDOW,
        ),
    ),
]
Executing action: ConfigureXlibWindow(
    Window {
        FIREFOX WINDOW
    },
)
Executing action: WindowTakeFocus {
    window: Window {
            FIREFOX WINDOW
    },
    previous_window: Some(
        Window {
            ALACRITTY WINDOW
        },
    ),
}
```

**Comment by carrascomj** [comment]
Oh, I have `focus_behavior="Sloppy"`, that may help to reproduce it.

By the way, it is not just firefox for me. Slack and teams also take the focus in the background upon spawning (and I think any window, I can't confirm now).

**Comment by AethanFoot** [comment]
Can you check #761, I think this may have been an issues with the urgency, and specifically firefox calling for urgency and taking focus (maybe as I can reproduce it on that branch).

**Comment by Zwo1in** [comment]
In my case #761 fixed this

**Comment by hertg** [comment]
closed via #761
